### PR TITLE
Mention agentless in the OpenSSH guide for better SEO

### PR DESCRIPTION
**Pull request by r0mant**
Right now searching for "agentless" in the docs does not bring up any results. This PR updates the OpenSSH guide to mention agentless mode explicitly so it shows up in the search results.

I have also removed the word "legacy" from "OpenSSH systems" because it feels like it has a bit of a negative/condescending connotation to it which I'm not sure we want to imply.

### CODE REVIEW

**Comment by ptgott** [comment]
Since the term "agent" is so prevalent in the docs, the Teleport source, and the way we talk about Teleport, I'm not sure we want to make the term "agentless mode" more visible. I'm worried that it will create some confusion around how Teleport works, since it's possible to make statements like, "You can run an SSH Service agent to enroll a server in agentless mode".

At the same time, the StrongDM website _still_ says you need to deploy a Teleport binary on every resource you want to proxy (https://www.strongdm.com/blog/strongdm-vs-teleport), so it would be nice if we had a way of communicating that Teleport is "agentless" in the StrongDM sense while still having "agents" in our own sense.

**Comment by lsgunnlsgunn** [comment]
Maybe we could lean into service as a term instead of agent and agentless. For example, something like: the Teleport service can manage access to resources without without being installed!running on those resources/hosts. 

Just a thought from the newbie.

**Comment by r0mant** [comment]
> Since the term "agent" is so prevalent in the docs, the Teleport source, and the way we talk about Teleport, I'm not sure we want to make the term "agentless mode" more visible. I'm worried that it will create some confusion around how Teleport works, since it's possible to make statements like, "You can run an SSH Service agent to enroll a server in agentless mode".

@ptgott I agree that "run Teleport agent in agentless mode" is confusing but if you notice I don't say that anywhere. "Running Teleport in agentless mode" makes sense IMO. Right now agentless does not bring any results in search, we should make it more visible.

**Comment by r0mant** [comment]
> Maybe we could lean into service as a term instead of agent and agentless. For example, something like: the Teleport service can manage access to resources without without being installed!running on those resources/hosts.

@lsgunnlsgunn Yep, we've gone through a very similar train of thought multiple times in the past with different variations of "agent", "service", "process", etc. :) The issue with "service" is that this term is a bit overloaded - Teleport has individual "application", "database", "ssh" and other services which all can run within the same process so it quickly gets confusing. Agent is a pretty standard term in the industry.

**Comment by ptgott** [comment]
@r0mant In "Running Teleport in agentless mode", does "Teleport" refer to the SSH Service, all Teleport services that proxy infrastructure resources, or the entire Teleport cluster?

**Comment by r0mant** [comment]
> In "Running Teleport in agentless mode", does "Teleport" refer to the SSH Service, all Teleport services that proxy infrastructure resources, or the entire Teleport cluster?

@ptgott I'm not sure any of these fit, it just refers to ability to use Teleport to connect to SSH nodes without installing Teleport agent on them. You still need proxy and auth running, and you still can have other agents (SSH or not) to connect to other resources.

**Comment by travelton** [comment]
My initial thought is to change the narrative and get away from **agent** and **agentless**, **node** is too ambiguous. 

I see everyone above using the term "connect", which describes what is happening under the hood. Perhaps a self-describing term like **connector**. When I think about other SaaS products with integrations, the typical way to describe a connection to internal resources is to use a connector.

In the context of deploying a Teleport Cloud tenant... IIRC, we’re going to support hosting the Kubernetes and Database nodes eventually. Additionally, we will host plugin binaries so customers don’t need to deploy and maintain the plugin binary, same for event export (fluentd, splunk, etc). All of these things could easily be referred to as connectors. I'd also recommend not calling it `Database Connector` or `Kubernetes Connector`... It's simply a `Teleport Connector` which supports one or all services. Teleport Proxy/Auth acts as the grand central station for all connectors and their connections.

- “Deploy a connector anywhere to access your Linux Server resources… In our cloud, in your cloud, or directly on the Linux Server itself.”

- “Deploy a connector anywhere to export events from Teleport into your SIEM of choice. In our cloud, in your cloud, or directly on the SIEM server itself.”

- “With Teleport’s flexible networking model and end-to-end encryption, a connector can securely exist anywhere. In our cloud, in your cloud, or directly on the server itself.”

I'll continue thinking about this and add more as I have time to collect my thoughts.

**Comment by r0mant** [comment]
While you guys are thinking about it, can we merge this? :) It's a simple change, it was not meant to turn into a refactor of our documentation and rethinking of the terms we use.

File: docs/pages/server-access/guides/openssh.mdx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+description: This guide shows you how to set up Teleport in agentless mode to enable secure access to OpenSSH servers so you can protect systems that do not run a Teleport binary.
+keywords: [openssh, teleport, agentless]
 videoBanner: x0eYFUEIOrM
 ---
 
-In this guide, we will show you how to configure the OpenSSH server `sshd` to
-join a Teleport cluster. Existing fleets of OpenSSH servers can be configured to
-accept SSH certificates dynamically issued by a Teleport CA.
+In this guide, we will show you how to configure Teleport in agentless mode and
+have the OpenSSH server `sshd` to join a Teleport cluster. Existing fleets of
```
Comment (alexfornuto) [inline docs/pages/server-access/guides/openssh.mdx:9]: Either:

```suggestion
configure the OpenSSH server `sshd` to join a Teleport cluster. Existing fleets of
```

or 

```suggestion
have the OpenSSH server `sshd` join a Teleport cluster. Existing fleets of
```

Comment (alexfornuto) [inline docs/pages/server-access/guides/openssh.mdx:9]: Either:

```suggestion
configure the OpenSSH server `sshd` to join a Teleport cluster. Existing fleets of
```

or 

```suggestion
have the OpenSSH server `sshd` join a Teleport cluster. Existing fleets of
```

**Review by zmb3 — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Pull request merged**